### PR TITLE
Fix #1917 Allow user to add custom headers to client

### LINKED ISSUE #1917: Allow user to add custom header for k8s requests?
Hey,  
I encountered the following problem：

When I init a client instance, I want to add  custom headers  for k8s request(such as userpin,cluster-id etc...),  ,  but Config object is only allow  to specify "masterUrl" , how i can add clustom header?  

Only by add interceptor ？  for example: 

```
        Config config = new ConfigBuilder().withMasterUrl(host).build();
        KubernetesClient client = new DefaultKubernetesClient(config);
        ((DefaultKubernetesClient) client).getHttpClient().networkInterceptors().add(chain -> {
            Request agent = chain.request().newBuilder().addHeader("user-pin","xxx")
                    .addHeader("cluster-id","xxxx").build();
            return chain.proceed(agent);
        });
```

Is it possible to set the customer header for requests sent to ones Kubernetes cluster in this API?

such as: 

```
Config config = new ConfigBuilder().withMasterUrl(northHost).withHeaders("name", "value").build();

```

@jimmidyson @iocanel @rohanKanojia

Thanks !

### PR DESCRIPTION
**Pull request by wangyushuai**
Fix #1917: Allow user to add custom headers to client

### CODE REVIEW

**Comment by oscerd** [comment]
please resolve the conflict on changelog. Other than that, it looks good

**Comment by rohanKanojia** [comment]
ok to test

**Comment by rohanKanojia** [comment]
@wangyushuai : Thanks for your PR. Could you please add license headers  using `mvn -N license:format` and resolve conflicts in `CHANGELOG.md`?

**Comment by wangyushuai** [comment]
> `mvn -N license:format`



> @wangyushuai : Thanks for your PR. Could you please add license headers using `mvn -N license:format` and resolve conflicts in `CHANGELOG.md`?

 add license headers using `mvn -N license:format`  ,  Sorry, I don't know how to do it ? Can you tell me how to do it？ 

@rohanKanojia

**Comment by rohanKanojia** [comment]
@wangyushuai : I have specified the command you need to run in order to add headers

**Comment by wangyushuai** [comment]
> @wangyushuai : I have specified the command you need to run in order to add headers

Get！ Thanks~ @rohanKanojia

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/HttpClientUtils.java [hunk 1]
```diff
@@ -183,6 +184,16 @@ private static OkHttpClient createHttpClient(final Config config, final Consumer
                 additionalConfig.accept(httpClientBuilder);
             }
 
+          if (config.getCustomHeaders() != null && !config.getCustomHeaders().isEmpty()) {
+            httpClientBuilder.addNetworkInterceptor(chain -> {
```
Comment (rohanKanojia) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/HttpClientUtils.java]: I think it should be `httpClientBuilder.addInterceptor(`, see line 115 where some other `Interceptors` are added.

Comment (rohanKanojia) [reply kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/HttpClientUtils.java]: No, both `networkInterceptor` and `Interceptor` are basically a `List<Interceptor>`. Seems like there was some problem in my setup due to which I couldn't see headers being applied. Now I can see them. Sorry for confusion :laughing:

**Comment by rohanKanojia** [comment]
[merge]

**Comment by rohanKanojia** [comment]
[merge]

**Comment by rohanKanojia** [comment]
@wangyushuai : Your fix is now available in [v4.7.1](https://repo1.maven.org/maven2/io/fabric8/kubernetes-client/4.7.1/)

**Comment by wangyushuai** [comment]
> @wangyushuai : Your fix is now available in [v4.7.1](https://repo1.maven.org/maven2/io/fabric8/kubernetes-client/4.7.1/)

ok， thanks you very much ！

**Pull request merged**